tests/s3/test_object_workflow_operations.py

```diff
@@ -727,3 +765,184 @@ def test_delete_objects_and_mention_1000_objects_7657(self):
+        self.log.info(
+            "ENDED: Test GET operation for object with key name consisting "
+            "of Hyphen(-)")
+
+    @pytest.mark.parallel
+    @pytest.mark.s3_ops
+    @pytest.mark.s3_object_ops
+    @pytest.mark.tags("TEST-45512")
+    def test_45512(self):
+        """Test GET operation for object with key name consisting of Period(.)"""
+        self.log.info(
+            "STARTED: Test GET operation for object with key name consisting "
+            "of Period(.)")
+        obj_key_name = [".objworkflow", "objworkflow.", ".objworkflow.", "obj.workflow"]
+        self.create_bucket_put_get_object(
+            self.bucket_name, obj_key_name, self.file_path,
+            S3_OBJ_TST["s3_object"]["mb_count"])
+        self.log.info(
+            "ENDED: Test GET operation for object with key name consisting "
+            "of Period(.)")
+
+    @pytest.mark.parallel
+    @pytest.mark.s3_ops
+    @pytest.mark.s3_object_ops
+    @pytest.mark.tags("TEST-45513")
+    def test_45513(self):
+        """Test GET operation for object with key name consisting of Single quote(')"""
+        self.log.info(
+            "STARTED: Test GET operation for object with key name consisting "
+            "of Single quote(')")
+        obj_key_name = ["'objworkflow", "objworkflow'", "'objworkflow'", "obj'workflow"]
+        self.create_bucket_put_get_object(
+            self.bucket_name, obj_key_name, self.file_path,
+            S3_OBJ_TST["s3_object"]["mb_count"])
+        self.log.info(
+            "ENDED: Test GET operation for object with key name consisting "
+            "of Single quote(')")
+
+    @pytest.mark.parallel
+    @pytest.mark.s3_ops
+    @pytest.mark.s3_object_ops
+    @pytest.mark.tags("TEST-45514")
+    def test_45514(self):
+        """Test GET operation for object with key name consisting of Asterisk(*)"""
+        self.log.info(
+            "STARTED: Test GET operation for object with key name consisting "
+            "of Asterisk(*)")
+        obj_key_name = ["*objworkflow", "objworkflow*", "*objworkflow*", "obj*workflow"]
+        self.create_bucket_put_get_object(
+            self.bucket_name, obj_key_name, self.file_path,
+            S3_OBJ_TST["s3_object"]["mb_count"])
+        self.log.info(
+            "ENDED: Test GET operation for object with key name consisting "
+            "of Asterisk(*)")
+
+    @pytest.mark.parallel
+    @pytest.mark.s3_ops
+    @pytest.mark.s3_object_ops
+    @pytest.mark.tags("TEST-45515")
+    def test_45515(self):
+        """Test GET operation for object with key name consisting of alphanumeric chars
+        and special chars with prefix and delimiter"""
+        self.log.info(
+            "STARTED: Test GET operation for object with key name consisting of "
+            "alphanumeric chars and special chars with prefix and delimiter")
+        obj_key_name = [
+            "My.family_photos-2014/jan/myvacation1.jpg", "videos/2014/birthday/video1.wmv"]
```

Comment:
Generate random string and then randomly insert `/`

Reply:
done